test/JDBC/expected/string_agg-vu-verify.out

```diff
@@ -0,0 +1,1352 @@
+Rohit               ?    Bhagat              
+~~END~~
+
+
+DECLARE @sep NVARCHAR(10) = N'🫥'
+SELECT STRING_AGG(col_char, @sep) FROM babel_5156_string_agg_t1 GROUP BY group_id ORDER BY group_id
+GO
+~~START~~
+varchar
+abc                 ??defghi             ???                  
+??·?                ???                  
+<NULL>
+Rohit               ?Bhagat              
+~~END~~
+
+
+DECLARE @sep CHAR(5) = N'尸'
+SELECT STRING_AGG(col_char, @sep) FROM babel_5156_string_agg_t1 GROUP BY group_id ORDER BY group_id
+GO
+~~START~~
+varchar
+abc                 ?    ?defghi             ?    ??                  
+??·?                ?    ??                  
+<NULL>
+Rohit               ?    Bhagat              
+~~END~~
+
+
+DECLARE @sep VARCHAR(10) = N'尸'
+SELECT STRING_AGG(col_char, @sep) FROM babel_5156_string_agg_t1 GROUP BY group_id ORDER BY group_id
+GO
+~~START~~
+varchar
+abc                 ??defghi             ???                  
+??·?                ???                  
+<NULL>
+Rohit               ?Bhagat              
+~~END~~
+
+
+DECLARE @sep NCHAR(5) = N'尸'
+SELECT STRING_AGG(col_char, @sep) FROM babel_5156_string_agg_t1 GROUP BY group_id ORDER BY group_id
+GO
+~~START~~
+varchar
+abc                 ?    ?defghi             ?    ??                  
+??·?                ?    ??                  
+<NULL>
+Rohit               ?    Bhagat              
+~~END~~
+
+
+DECLARE @sep NVARCHAR(10) = N'尸'
+SELECT STRING_AGG(col_char, @sep) FROM babel_5156_string_agg_t1 GROUP BY group_id ORDER BY group_id
+GO
+~~START~~
+varchar
+abc                 ??defghi             ???                  
+??·?                ???                  
+<NULL>
+Rohit               ?Bhagat              
+~~END~~
+
+
+-- input type varchar
+SELECT STRING_AGG(col_varchar, NULL) FROM babel_5156_string_agg_t1 GROUP BY group_id ORDER BY group_id
+GO
+~~START~~
+varchar
+abc?defghi??
+??·???
+<NULL>
+RohitBhagat
+~~END~~
+
+
+SELECT STRING_AGG(col_varchar, '-') FROM babel_5156_string_agg_t1 GROUP BY group_id ORDER BY group_id
+GO
+~~START~~
+varchar
+abc-?defghi-??
+??·?-??
+<NULL>
+Rohit-Bhagat
+~~END~~
+
+
+DECLARE @sep CHAR(1) = '-'
+SELECT STRING_AGG(col_varchar, @sep) FROM babel_5156_string_agg_t1 GROUP BY group_id ORDER BY group_id
+GO
+~~START~~
+varchar
+abc-?defghi-??
+??·?-??
+<NULL>
+Rohit-Bhagat
+~~END~~
+
+
+DECLARE @sep VARCHAR(10) = '-'
+SELECT STRING_AGG(col_varchar, @sep) FROM babel_5156_string_agg_t1 GROUP BY group_id ORDER BY group_id
+GO
+~~START~~
+varchar
+abc-?defghi-??
+??·?-??
+<NULL>
+Rohit-Bhagat
+~~END~~
+
+
+DECLARE @sep NCHAR(1) = '-'
+SELECT STRING_AGG(col_varchar, @sep) FROM babel_5156_string_agg_t1 GROUP BY group_id ORDER BY group_id
+GO
+~~START~~
+varchar
+abc-?defghi-??
+??·?-??
+<NULL>
+Rohit-Bhagat
+~~END~~
+
+
+DECLARE @sep NVARCHAR(10) = '-'
+SELECT STRING_AGG(col_varchar, @sep) FROM babel_5156_string_agg_t1 GROUP BY group_id ORDER BY group_id
+GO
+~~START~~
+varchar
+abc-?defghi-??
+??·?-??
+<NULL>
+Rohit-Bhagat
+~~END~~
```

Comment:
Document mentions that `If the input expression is type VARCHAR, the separator cannot be type NVARCHAR.`, we should handle this case as well

Reply:
Added check to handle this scenario.